gdk/commands/component/InitCommand.py

```diff
@@ -0,0 +1,135 @@
+
+        Parameters
+        ----------
+          command_args(dict): A dictionary object that contains parsed args namespace of a command.
+
+        Returns
+        -------
+            None
+        """
+        project_dir = utils.current_directory
+        if not self.command_args["name"]:
+            # Check if directory is not empty
+            if not utils.is_directory_empty(project_dir):
+                raise Exception(error_messages.INIT_NON_EMPTY_DIR_ERROR)
+        else:
+            # Create a new directory with name.
+            project_dir = Path(project_dir).joinpath(self.command_args["name"]).resolve()
+            try:
+                logging.debug("Creating new project directory '{}' in the current directory.".format(project_dir.name))
+                Path(project_dir).mkdir(exist_ok=False)
+            except FileExistsError:
+                raise Exception(error_messages.INIT_DIR_EXISTS_ERROR.format(project_dir.name))
+
+        # Choose appropriate action based on commands
+        if self.command_args["template"] and self.command_args["language"]:
+            template = self.command_args["template"]
+            language = self.command_args["language"]
+            if template and language:
+                logging.info(
+                    "Initializing the project directory with a {} component template - '{}'.".format(language, template)
+                )
+                self.init_with_template(template, language, project_dir)
+                return
+        elif self.command_args["repository"]:
+            repository = self.command_args["repository"]
+            if repository:
+                logging.info(
+                    "Initializing the project directory with a component from repository catalog - '{}'.".format(repository)
+                )
+                self.init_with_repository(repository, project_dir)
+                return
+        raise Exception(error_messages.INIT_WITH_INVALID_ARGS)
+
+    def init_with_template(self, template, language, project_dir):
+        try:
+            template_name = "{}-{}".format(template, language)
+            logging.info("Fetching the component template '{}' from Greengrass Software Catalog.".format(template_name))
+            self.download_and_clean(template_name, "template", project_dir)
+        except Exception as e:
+            raise Exception("Could not initialize the project with component template '{}'.\n{}".format(template, e))
+
+    def init_with_repository(self, repository, project_dir):
+        try:
+            logging.info("Fetching the component repository '{}' from Greengrass Software Catalog.".format(repository))
+            self.download_and_clean(repository, "repository", project_dir)
+        except Exception as e:
+            raise Exception("Could not initialize the project with component repository '{}'.\n{}".format(repository, e))
+
+    def download_and_clean(self, comp_name, comp_type, project_dir):
```

Comment:
This download function is generic and could be move into a utility

Reply:
Right but we use it only with the `init` command.